spec/app/db/repository/authenticator_repository_spec.rb

```diff
@@ -3,207 +3,221 @@
 require 'spec_helper'
 
 RSpec.describe('DB::Repository::AuthenticatorRepository') do
-  let(:resource_repository) { ::Resource }
-
-  let(:repo) do
-    DB::Repository::AuthenticatorRepository.new(
-      resource_repository: resource_repository,
-      data_object: Authentication::AuthnOidc::V2::DataObjects::Authenticator
-    )
-  end
-
-  let(:arguments) { %i[provider_uri client_id client_secret claim_mapping nonce state] }
+  %w[true false].each do |enabled|
+    context "with flag #{enabled}" do
+      with_feature_enabled(:pkce_support) do
+        let(:data_object) do
+          if Rails.configuration.feature_flags.enabled?(:pkce_support)
+            Authentication::AuthnOidc::PkceSupportFeature::DataObjects::Authenticator
+          else
+            Authentication::AuthnOidc::V2::DataObjects::Authenticator
+          end
```

Comment:
Here's an example of where I'd rather see this driven by test inputs, the `enabled` option above, rather than indirectly through the feature flag environment variable.